src/RemoteInboxNotifications/RuleEvaluator.php

```diff
@@ -42,15 +45,33 @@ public function evaluate( $rules, $stored_state = null ) {
 			return false;
 		}
 
+		$evaluation_logger = null;
+
+		if ( count( $logger_args ) ) {
+			if ( ! array_key_exists( 'slug', $logger_args ) ) {
+				throw new \InvalidArgumentException( 'Missing required field: slug in $logger_args.' );
+			}
+
+			array_key_exists( 'source', $logger_args ) ? $source = $logger_args['source'] : $source = null;
+
+			$evaluation_logger = new EvaluationLogger( $logger_args['slug'], $source );
+		}
+
 		foreach ( $rules as $rule ) {
 			$processor        = $this->get_rule_processor->get_processor( $rule->type );
 			$processor_result = $processor->process( $rule, $stored_state );
+			$evaluation_logger && $evaluation_logger->add_result( $rule->type, $processor_result );
 
 			if ( ! $processor_result ) {
 				return false;
 			}
 		}
 
+		defined( 'WC_ADMIN_DEBUG_RULE_EVALUATOR' )
```

Comment:
Maybe this check should be moved inside of the `log()` method to avoid duplication.  This way we can just call `log()` anywhere and let it handle whether or not `WC_ADMIN_DEBUG_RULE_EVALUATOR` is defined.

Reply:
@joshuatf That's a good idea 👍 I'll make a change.

Reply:
@joshuatf updated in https://github.com/woocommerce/woocommerce-admin/pull/7194/commits/323c9fad78f530c07fb14cfe882d31388b7c1bef